overlord/snapstate/snapmgr.go

```diff
@@ -227,23 +250,39 @@ func Manager(s *state.State) (*SnapManager, error) {
 	return m, nil
 }
 
-// Store returns the store service used by the manager.
-func (m *SnapManager) Store() StoreService {
-	return m.store
+// ReplaceStore replaces the store used by the manager.
+func ReplaceStore(state *state.State, store StoreService) {
+	state.Cache("store", store)
 }
 
-// ReplaceStore replaces the store used by manager.
-func (m *SnapManager) ReplaceStore(store StoreService) {
-	m.store = store
+func Store(s *state.State) StoreService {
+	if ubuntuStore := s.Cached("store"); ubuntuStore != nil {
```

Comment:
to avoid clashes,  we want something like:

type cacheKey string

and cacheKey("store")